cli/tests/test_log_command.rs

```diff
@@ -179,7 +179,7 @@ fn test_log_with_or_without_diff() {
     insta::assert_snapshot!(stderr, @r###"
     error: the argument '--git' cannot be used with '--color-words'
 
-    Usage: jj log --template <TEMPLATE> --no-graph --patch --git [PATHS]...
+    Usage: jj log --template [<TEMPLATE>] --no-graph --patch --git [PATHS]...
```

Comment:
It's unfortunate that the `--template` value has to be optional.

Another implementation idea is to inspect the `clap::Error` at `handle_command_result()` or `CliRunner::run_internal()`.
```
[cli/src/command_error.rs:478:13] (inner.kind(), inner.context().collect_vec()) = (
    InvalidValue,
    [
        (
            InvalidArg,
            String(
                "--template <TEMPLATE>",
            ),
        ),
        (
            InvalidValue,
            String(
                "",
            ),
        ),
        (
            ValidValue,
            Strings(
                [],
            ),
        ),
    ],
)
```

Doing that seems intrusive, but I think it's okay because `--template` is the concept apply to many commands. A downside is that we'll need to look up template aliases from the config object (because `WorkspaceCommandHelper` isn't instantiated yet.

Reply:
I've implemented this variant in #3200. Please take a look.